actions/typescript-test/action.yml

```diff
@@ -0,0 +1,30 @@
+name: "Perform a code test"
+description: "Installing dependenncies an running a test"
+
+inputs:
+   token:
+     description: "Token needed to access files "
+     required: true
+runs:
+   using: "composite"
+   steps:
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '14'
+      - name: Cache NPM dependencies
+        uses: actions/cache@v2
+        with:
+          path: ~/.npm
+          key: ${{ runner.OS }}-npm-cache-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.OS }}-npm-cache-
```

Comment:
Shouldn't you try to restore the exact same key? I assume there might be multiple caches containing keys such `linux-npm-cache-hash123`, `linux-npm-cache-hash456` .. if you use `linux-npn-cache-` only for restoring I am not sure you will get the right cache . 
I might be wrong about it as I remember how this works in CircleCI but I am not sure in github actions but I assume it will be similar.